### PR TITLE
chore(web-modeler): adjust env variables for OIDC configuration scheme

### PR DESCRIPTION
**Pull request by wollefitz**
Part of https://github.com/camunda/web-modeler/issues/7077

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -5,7 +5,7 @@ CAMUNDA_PLATFORM_VERSION=8.3.4
 # renovate: datasource=docker depName=camunda/optimize
 CAMUNDA_OPTIMIZE_VERSION=8.3.3
 # renovate: datasource=docker depName=camunda/web-modeler lookupName=registry.camunda.cloud/web-modeler-ee/modeler-restapi
-CAMUNDA_WEB_MODELER_VERSION=8.3.2
+CAMUNDA_WEB_MODELER_VERSION=8.4.0-alpha2
```
[the remark this answers is not in the document]
Comment (jfriedenstab) [reply .env:8]: I'm not sure either; but I'm wondering why `main` hasn't been updated to `8.4.0-alpha1` already with the last release.

@camunda/distribution: What's our policy here?

[the remark this answers is not in the document]
Comment (jessesimpson36) [reply .env:8]: Perhaps we need to be more specific in our documentation about our release policy for this repo:
https://confluence.camunda.com/display/HAN/Release+Policy+-+Distribution

> Alpha: Supported in docker-compose, later in helm-charts (images can be set manually for now).

We don't state explicitly what should be in main for docker-compose, just that alpha releases are intended to be supported. 

I think you're right, it's intended to be the alpha version.

File: docker-compose-web-modeler.yaml [hunk 2]
```diff
@@ -125,11 +125,9 @@ services:
       CLIENT_PUSHER_FORCE_TLS: "false"
       CLIENT_PUSHER_KEY: modeler-app-key
       OAUTH2_CLIENT_ID: web-modeler
+      OAUTH2_JWKS_URL: http://keycloak:8080/auth/realms/camunda-platform/protocol/openid-connect/certs
       OAUTH2_TOKEN_AUDIENCE: web-modeler
       OAUTH2_TOKEN_ISSUER: http://localhost:18080/auth/realms/camunda-platform
-      KEYCLOAK_BASE_URL: http://localhost:18080
-      KEYCLOAK_REALM: camunda-platform
```
Comment (jessesimpson36) [inline docker-compose-web-modeler.yaml:131]: is KEYCLOAK_REALM not necessary anymore?

File: docker-compose-web-modeler.yaml [hunk 3]
```diff
@@ -76,6 +76,7 @@ services:
     environment:
       JAVA_OPTIONS: -Xmx128m
       LOGGING_LEVEL_IO_CAMUNDA_MODELER: DEBUG
+      CAMUNDA_IDENTITY_BASE_URL: http://identity:8084/
```
Comment (jfriedenstab) [inline docker-compose-web-modeler.yaml:79]: Given the configuration property `camunda.identity.base-url` I think the [recommended format](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.external-config.typesafe-configuration-properties.relaxed-binding.environment-variables) for the environment variable would be `CAMUNDA_IDENTITY_BASEURL` (it's also used in the C8 [Helm chart](https://github.com/camunda/camunda-platform-helm/blob/e5d5dbcdba68692a50e31bcd41ffff6d30d25de7/charts/camunda-platform/templates/camunda/configmap-identity-auth.yaml#L9)). However, the variant with the underscore seems to work fine as well.

**Review by jfriedenstab — APPROVED** [review]

File: docker-compose-web-modeler.yaml (hunk 2 above)
Comment (jfriedenstab) [reply docker-compose-web-modeler.yaml:131]: @jessesimpson36: We're using a new library for the authentication now that doesn't require the realm to be configured separately any more. It's already part of `OAUTH2_TOKEN_ISSUER` and `OAUTH2_JWKS_URL` anyway.

File: .env (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfriedenstab) [reply .env:8]: This topic was discussed on Slack: https://camunda.slack.com/archives/C01H4NG9XDY/p1702541447938269

**Pull request merged**